platform-server-rest/src/main/java/uno/cod/platform/server/rest/controller/ChallengeController.java

```diff
@@ -17,21 +18,19 @@
 public class ChallengeController {
 
     private final ChallengeService service;
-    private final ChallengeTemplateService challengeTemplateService;
 
     @Autowired
-    public ChallengeController(ChallengeService challengeService, ChallengeTemplateService challengeTemplateService){
+    public ChallengeController(ChallengeService challengeService){
         this.service = challengeService;
-        this.challengeTemplateService = challengeTemplateService;
     }
 
     @RequestMapping(value = RestUrls.CHALLENGES_ID, method = RequestMethod.GET)
-    @PreAuthorize("isAuthenticated() and @securityService.canAccessScheduledChallengeChallenge(principal, #id)")
-    public ResponseEntity<ChallengeTemplateShowDto> get(@PathVariable Long id) {
-        return new ResponseEntity<>(challengeTemplateService.findByChallengeId(id), HttpStatus.OK);
+    @PreAuthorize("isAuthenticated() and @securityService.canAccessChallenge(principal, #id)")
+    public ResponseEntity<ChallengeDto> get(@PathVariable Long id){
+        return new ResponseEntity<>(service.findOneById(id), HttpStatus.CREATED);
     }
 
-    @RequestMapping(value = RestUrls.CHALLENGE_TEMPLATES_CHALLENGE, method = RequestMethod.POST)
+    @RequestMapping(value = RestUrls.CHALLENGE_TEMPLATES__ID_CHALLENGE, method = RequestMethod.POST)
```

Comment:
why not post on `/challenges` like a normal rest route and have the `templateId` in the `dto`?